PRODUCTNAME/Dangerfile

```
@@ -33,3 +35,43 @@ swiftlint.config_file = "#{src_root}/.swiftlint.yml"
 # Run Swift-Lint and warn us if anything fails it
 swiftlint.directory = src_root
 swiftlint.lint_files inline_mode: true
+
+# Getting artifact URLs from CircleCI
+
+# Set this yourself using the steps in "Getting Started"
+token = ENV['CIRCLE_API_TOKEN']
+# These are already in the Circle environment
+# https://circleci.com/docs/2.0/env-vars/#build-specific-environment-variables
+username = ENV['CIRCLE_PROJECT_USERNAME']
+reponame = ENV['CIRCLE_PROJECT_REPONAME']
+build = ENV['CIRCLE_BUILD_NUM']
+
+if !token.nil? and !username.nil? and !reponame.nil? and !build.nil?
+	fetcher = CircleciArtifact::Fetcher.new token, username, reponame, build
+
+	xcov = CircleciArtifact::Query.new 'xcov/index.html'
+	slather = CircleciArtifact::Query.new 'slather/index.html'
+	screenshots = CircleciArtifact::Query.new 'screenshots/index.html'
+	queries = [xcov, slather, screenshots]
+	results = fetcher.fetch(queries)
+
+	xcov_url = results.url_for_query(xcov)
+	slather_url = results.url_for_query(slather)
+	screenshots_url = results.url_for_query(screenshots)
+
+	if !xcov_url.nil?
+		message "[xcov](#{xcov_url})"
+	end
+
+	if !slather_url.nil?
+		message "[Slather](#{slather_url})"
+	end
+
+	if !screenshots_url.nil?
+		message "[Screenshots](#{screenshots_url})"
+	end
+else
+	warn "Missing CircleCI artifacts. [CIRCLE_API_TOKEN](https://github.com/Raizlabs/circleci_artifact#getting-started) not set."
```

#nth this error message applies only if the token is nil, but three other things could contribute to its being fired. Might want to add more specific error messages. I don't know what Ruby has in terms of bailing early, but if there's something like Swift's `guard` so you can throw an error message per error, that might be good.

---

Since Danger is being run on CircleCI, everything but the CIRCLE_API_TOKEN is always in the environment by default. Will improve the message